test/src/test/java/hudson/model/queue/QueueTaskDispatcherTest.java

```diff
@@ -57,14 +58,19 @@ public String getShortDescription() {
     public void canTakeBlockageIsDisplayed() throws Exception {
         FreeStyleProject project = r.createFreeStyleProject();
         r.jenkins.getQueue().schedule(project);
-        Queue.Item item;
-        while (true) {
+        Queue.Item item = null;
+        long timeout = TimeUnit.MINUTES.toMillis(1);
+        long started = System.currentTimeMillis();
+
+        while (System.currentTimeMillis() - started < timeout) {
             item = r.jenkins.getQueue().getItem(project);
-            if (item.isBuildable()) {
+            if (item.isBuildable() && item.getCauseOfBlockage().getShortDescription().contains("blocked by canTake")) {
```

Comment:
Ideally there should be a null check of `getCauseOfBlockage()`

Reply:
In current implementation `item.isBuildable` does the same (implicitly), because `Buildable .getCauseOfBlockage()` never returns null. But I can add null check as well, of course.